component/s3/file/s3_test.go

```diff
@@ -0,0 +1,76 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/johannesboyne/gofakes3"
+	"github.com/johannesboyne/gofakes3/backend/s3mem"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCorrectBucket(t *testing.T) {
+	o := component.Options{
+		ID:            "t1",
+		OnStateChange: func(_ component.Exports) {},
+	}
+	s3File, err := New(o,
+		Arguments{
+			Path:          "s3://bucket/file",
+			PollFrequency: 30 * time.Second,
+			IsSecret:      false,
+		})
+	require.NoError(t, err)
+	require.NotNil(t, s3File)
+}
+
+func TestWatchingFile(t *testing.T) {
+	_, srv := pushFilesToFakeS3(t, "test.txt", "success!")
+	s3File, err := New(component.Options{
+		ID:            "id1",
+		OnStateChange: func(_ component.Exports) {},
+	}, Arguments{
+		Path:          "s3://mybucket/test.txt",
+		PollFrequency: 100 * time.Millisecond,
+		IsSecret:      false,
+		Options: AWSOptions{
+			Endpoint:     srv.URL,
+			DisableSSL:   true,
+			UsePathStyle: true,
+		},
+	})
+	require.NoError(t, err)
+	ctx, cancel := context.WithCancel(context.Background())
+	go s3File.Run(ctx)
+	time.Sleep(200 * time.Millisecond)
```

Comment:
I low-key hate this sleep. Really low poll and sleep values caused it to be inconsistent in passing though.